OPAL/ai/src/main/scala/org/opalj/ai/domain/l1/ClassValues.scala

```diff
@@ -184,8 +189,23 @@ trait ClassValues
                             )
                     }
 
+                case _ if secondOperandIsStringValue =>
```

Comment:
I wonder if this matching could be done in a nicer way, maybe along the lines of
```
operands match {
  case _ :+ (sv: StringValue) =>
  case _ :+ (sv: StringValue) :+ _ =>
}
```
?

Reply:
Or maybe do away with the value match and just match on the descriptor?

Reply:
I now changed this so it uses `operands.reverse match { ... }` - should essentially be the same.

Reply:
It creates a reversed list, though. Probably not a performance issue here, though.

Reply:
Ah yes that's right - i changed that to your suggestion. Thought the reversed version would be slightly more readable, as the first operand would actually be the first in the match clauses, but i think performance should be more important here. Could you re-approve your review so i can merge?

Reply:
Did you make sure it works?

Reply:
Well i ran all the new unit tests and it does yield the desired results - it seems to be functionally equivalent to the previous solutions. Comments for readability are already in place, so i think we're fine there.